src/proto/generation.proto

```diff
@@ -242,6 +244,100 @@ message ClassifierParameters {
   optional Action realized_action = 3;
 }
 
+
+// Interpolation using intermediate flows computed from first two images and
+// applied at specified ratios to either those or the following images.
+message InterpolateParameters {
+  repeated float ratios = 1;
+}
+
+
+//
+// Transforms
+//
+
+enum BorderMode {
+  BORDER_REFLECT = 0;   // reflect image values across the border
+  BORDER_REPLICATE = 1; // replicate border values outside the image
+  BORDER_WRAP = 2;      // wrap around / tile the image values
+  BORDER_ZERO = 3;      // use 0 for locations outside the image
+}
+
+enum ColorMatchMode {
+  COLOR_MATCH_HSV = 0;  // match hue, saturation, and value histograms
+  COLOR_MATCH_LAB = 1;  // match lightness, a, and b histograms
+  COLOR_MATCH_RGB = 2;  // match red, green, and blue histograms
+}
+
+message TransformAddNoise {
+  float amount = 1;         // amount of gaussian noise to add
+  optional uint32 seed = 2; // random seed for the noise generation
+}
+
+message TransformBlend {
+  float amount = 1;     // blend amount from 0 to 1
+  Artifact target = 2;  // target image
+}
+
+message TransformColorMatch {
+  ColorMatchMode color_mode = 1;  // color mode to use
+  Artifact image = 2;             // image to match
+}
+
+message TransformContrast {
+  optional float brightness = 1;  // scale on pixel intensities, 1.0 is no change
+  optional float contrast = 2;    // contrast adjustment, 1.0 is no change
+}
+
+message TransformDepthCalc {
+  optional bool export = 1;        // return the depth map as an artifact
+  optional float blend_weight = 2; // blend factor between AdaBins (0.0) and MiDaS (1.0)
+}
+
+message TransformWarp2d {
+  BorderMode border_mode = 1; // extrapolation of border pixels
+  float translate_x = 4;      // horizontal translation in pixels
+  float translate_y = 5;      // vertical translation in pixels
+  float rotate = 2;           // rotation in degrees
+  float scale = 3;            // scale factor (>1 zooms in, <1 zooms out)
+}
+
+message TransformWarp3d {
+  BorderMode border_mode = 1; // extrapolation of border pixels
+  float translate_x = 3;      // horizontal translation
+  float translate_y = 4;      // vertical translation
+  float translate_z = 5;      // forward translation
+  float rotate_x = 6;         // rotation around X-axis in degrees (pitch)
+  float rotate_y = 7;         // rotation around Y-axis in degrees (yaw)
+  float rotate_z = 8;         // rotation around Z-axis in degrees (roll)
+  float near_plane = 9;       // near clipping plane of view frustum
+  float far_plane = 10;       // far clipping plane of view frustum
+  float fov = 11;             // field of view in degrees
+}
+
+message TransformWarpFlow {
```

Comment:
most of these operations don't need to be mask-aware, but I'm pretty sure this one does. if a mask prompt is provided, will the mask be available to the transform operations?